app/Models/Campaign.php

```diff
@@ -277,6 +277,39 @@ public function setArbitraryDateString($attribute, $value)
      */
     public function toSearchableArray()
     {
-        return Arr::except($this->toArray(), ['impact_doc']);
+        $array = $this->toArray();
+
+        // Append data from Action model relationship.
+        $array['actions'] = $this->actions->map(function ($data) {
+            return Arr::only($data->toArray(), [
+                'action_type',
+                'anonymous',
+                'civic_action',
+                'name',
+                'noun',
+                'online',
+                'post_type',
+                'quiz',
+                'reportback',
+                'scholarship_entry',
+                'time_commitment',
+                'verb',
+                'volunteer_credit',
+            ]);
+        });
+
+        // Only send data we want to search against.
+        return Arr::only($array, [
+            'accepted_count',
+            'actions',
+            'contentful_campaign_id',
+            'cause',
+            'end_date',
+            'id',
+            'internal_title',
+            'pending_count',
+            'secondary_causes',
+            'start_date',
+        ]);
```

Comment:
With the addition of the action relationship data, I switched to using the `only()` method instead of `except()` so it's easier at a glance to see exactly what we are sending to Algolia. Since Algolia has a cap for the size of each record, it's best to really only send exactly the data you want to be able to search on. While not all of these items seem relevant for search in Phoenix, we think it could help with Admins searching in Rogue in a future update!

Reply:
Love that switch, makes it way more obvious to see what is getting pulled in!